fuzz/fuzz_packfile_raw.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * libgit2 raw packfile fuzz target.
+ *
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include <openssl/sha.h>
+
+#include "git2.h"
+
+#define REPOSITORY_PATH "fuzz.git"
+
+static git_repository *repo = NULL;
+static git_odb *odb = NULL;
+
+int LLVMFuzzerInitialize(__attribute__((unused)) int *argc,
+			 __attribute__((unused)) char ***argv)
+{
+	if (git_libgit2_init() < 0) {
+		fprintf(stderr, "Failed to initialize libgit2\n");
+		abort();
+	}
+	if (git_repository_init(&repo, REPOSITORY_PATH, true) < 0) {
```

Comment:
I think instead of having a repository being created in the current directory, you should instead create a temporary path by using e.g. `mkdtemp`.

Reply:
How about I use `git_mempack_new` instead? No files except the `pack_git_...` temporaries which will only be leaked if the process crashes. I also added some code to remove the `*.pack` and `*.idx` files that are committed.

The problem with the fuzzers is that they are not designed to clean up the process state when they go down. Even using `atexit(3)` wasn't enough to clean the mess left.

Reply:
That's even better!